src/foam/flow/PromiseSlot.js

```diff
@@ -0,0 +1,33 @@
+foam.CLASS({
+  package: 'foam.flow',
+  name: 'PromiseSlot',
+  implements: [ 'foam.core.Slot' ],
+  documentation: `
+    A slot that takes a promise and sets its value to its value when it
+    resolves.
+  `,
+  properties: [
+    {
+      name: 'promise',
+      postSet: function(_, n) {
+        n.then(function(v) {
+          if ( n === this.promise ) this.value = v;
+        }.bind(this));
```

Comment:
We should also handle the error case of the promise I think.  Not sure what we should do, but I'm not sure we should just ignore it.

Reply:
Maybe its okay, since we ignore exceptions in most other places and a Promise rejection is basically an exception...  not sure

Reply:
Yeah not sure what to do either but I'm ok with ignoring this until we do think of a good way to handle it rather than rushing in behaviour that we might not end up keeping.